### ISSUE TITLE
Unable to read binlogs from MySql 5.7 using BinlogConnectorReplicator

### ISSUE THREAD

**Issue by agrawalrajneesh** [zendesk/maxwell#715]
I am trying to read binlogs from mysql 5.7. I am using BinlogConnectorReplicator by setting binlog_connector=true. But the execution is not going in while loop which is in listenForEventPackets() method of com.github.shyiko.mysql.binlog.BinaryLogClient class. 

private void listenForEventPackets() throws IOException {
        ByteArrayInputStream inputStream = channel.getInputStream();
        boolean completeShutdown = false;
        try {
            while (inputStream.peek() != -1) {
                int packetLength = inputStream.readInteger(3);
                inputStream.skip(1); // 1 byte for sequence
                int marker = inputStream.read();

"inputStream.peek() != -1" condition is never getting true. As A result I am not able to create Event. I am using v1.10.3 of Maxwell.

On the other side, If I am using com.google.code.or.OpenReplicator by setting binlog_connector=false, Everything is working fine and I am able to get event in kafka.

Please suggest me a resolution.

**Comment by shyiko** [comment]
@agrawalrajneesh try upgrading mysql-binlog-connector-java to 0.13.0. The problem might be connected to some of the issues fixed around 0.10.0 release but it's hard to tell. If it doesn't help - maxwell maintainers will have at least this thing to rule out. 

<sup>(I'm not the maintainer)</sup>

**Comment by agrawalrajneesh** [comment]
@shyiko Thanks, I will upgrade mysql-binlog-connector-java to 0.13.0.

**Comment by agrawalrajneesh** [comment]
I upgraded mysql-binlog-connector-java to 0.13.0. I am getting the exception

java.util.concurrent.TimeoutException: BinaryLogClient was unable to connect in 5000ms
	at com.github.shyiko.mysql.binlog.BinaryLogClient.connect(BinaryLogClient.java:819)
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.startReplicator(BinlogConnectorReplicator.java:107)
	at com.zendesk.maxwell.replication.BinlogConnectorReplicator.beforeStart(BinlogConnectorReplicator.java:113)
	at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:23)
	at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:186)
	at com.zendesk.maxwell.Maxwell.start(Maxwell.java:134)
	at com.zendesk.maxwell.Maxwell.main(Maxwell.java:206)

Just to let you know that I have commented gtid related code in com.github.shyiko.mysql.binlog.BinaryLogClient since it was throwing errors.
//private GtidSet gtidSet;
//private final Object gtidSetAccessLock = new Object();
//private boolean gtidSetFallbackToPurged;

If I use com.google.code.or.OpenReplicator by setting binlog_connector=false, It is working fine.

**Comment by shyiko** [comment]
> I have commented gtid related code in com.github.shyiko.mysql.binlog.BinaryLogClient since it was throwing errors.

What kind of error?

Also, to speed things up, can you please specify the exact version of MySQL server and share your my.conf (if you have a vagrant or docker image - that would be helpful too)?

**Comment by agrawalrajneesh** [comment]
**Mysql version**
root@rajneesh-VirtualBox:/home/rajneesh# mysql --version
mysql  Ver 14.14 Distrib 5.7.19, for Linux (x86_64) using  EditLine wrapper

**my.conf**
!includedir /etc/mysql/conf.d/
!includedir /etc/mysql/mysql.conf.d/
[mysqld]
server-id=12
log-bin = /var/lib/mysql/mysql-bin
binlog_format =ROW

**Comment by agrawalrajneesh** [comment]
I have uncommented gtid related code. The error is same.

java.util.concurrent.TimeoutException: BinaryLogClient was unable to connect in 5000ms
at com.github.shyiko.mysql.binlog.BinaryLogClient.connect(BinaryLogClient.java:819)
at

**Comment by agrawalrajneesh** [comment]
I noticed some errors on maxscale logs when I try to run Maxwell using binlog connector.

Aug 02 16:46:02 rajneesh-VirtualBox maxscale[7031]: Replication: attempting to connect to master server 127.0.0.1:3306, binlog mysql-bin.000006, pos 121693
Aug 02 16:46:02 rajneesh-VirtualBox maxscale[7031]: Replication: Request binlog records from mysql-bin.000006 at position 121693 from master server 127.0.0.1:3306
Aug 02 16:46:02 rajneesh-VirtualBox maxscale[7031]: Replication: identity seen by the master: server_id: 16, uuid: cc82a093-776b-11e7-8e87-080027bd7962
Aug 02 16:46:02 rajneesh-VirtualBox maxscale[7031]: Replication: identity seen by the slaves: server_id: 12, uuid: f6cdfc61-6d16-11e7-9f8c-080027bd7962, hostname: rajneesh
Aug 02 16:46:02 rajneesh-VirtualBox maxscale[7031]: **Replication: Master connection error 2003 '#HY000 Lost connection to backend server.' in state 'Binlog Dump', attemptin**
Aug 02 16:46:02 rajneesh-VirtualBox maxscale[7031]: Replication: Master 127.0.0.1 disconnected after 0 seconds. 0 events read.
Aug 02 16:46:02 rajneesh-VirtualBox maxscale[7031]: Replication: attempting to connect to master server 127.0.0.1:3306, binlog mysql-bin.000006, pos 121693
Aug 02 16:46:02 rajneesh-VirtualBox maxscale[7031]: Replication: Request binlog records from mysql-bin.000006 at position 121693 from master server 127.0.0.1:3306
Aug 02 16:46:02 rajneesh-VirtualBox maxscale[7031]: Replication: identity seen by the master: server_id: 16, uuid: cc82a093-776b-11e7-8e87-080027bd7962
Aug 02 16:46:02 rajneesh-VirtualBox maxscale[7031]: Replication: identity seen by the slaves: server_id: 12, uuid: f6cdfc61-6d16-11e7-9f8c-080027bd7962

I found a similar issue reported to maxscale for the same.
https://jira.mariadb.org/browse/MXS-1161
